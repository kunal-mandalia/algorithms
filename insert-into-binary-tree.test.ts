import insertIntoBinaryTree from './insert-into-binary-tree'

describe('insert into binary tree', () => {
  it('should insert node at max depth', () => {
    // arrange
    const tree = {
      value: 2,
      left: {
        value: 1,
        left: null,
        right: null
      },
      right: null
    }

    const node = {
      value: 3,
      left: null,
      right: null,
    }

    // act
    // assert
    expect((insertIntoBinaryTree(tree, node))).toMatchSnapshot()
  })

  it('should insert node between root and max level', () => {
    // arrange
    const tree = {
      value: 2,
      left: {
        value: 1,
        left: null,
        right: null
      },
      right: {
        value: 100,
        left: null,
        right: null
      }
    }

    const node = {
      value: 3,
      left: null,
      right: null,
    }

    /**
     *  1
     * 2 100
     * 
     *    1
     *  2,null  
     */

    // act
    // assert
    expect((insertIntoBinaryTree(tree, node))).toMatchSnapshot()
  })

  it('should insert node between root and max level (2)', () => {
    // arrange
    const tree = {
      value: 2,
      left: {
        value: 1,
        left: null,
        right: null
      },
      right: {
        value: 100,
        right: null,
        left: null
      }
    }

    const node = {
      value: 150,
      left: {
        value: 50,
        left: null,
        right: null
      },
      right: null,
    }

    /**
     *  1
     * 2 100
     * 
     *    1
     *  2,null  
     */

    // act
    // assert
    expect((insertIntoBinaryTree(tree, node))).toMatchSnapshot()
  })
})