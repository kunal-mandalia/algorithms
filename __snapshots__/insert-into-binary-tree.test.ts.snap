// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`insert into binary tree should insert node at max depth 1`] = `
Object {
  "left": Object {
    "left": null,
    "right": null,
    "value": 1,
  },
  "right": Object {
    "left": null,
    "right": null,
    "value": 3,
  },
  "value": 2,
}
`;

exports[`insert into binary tree should insert node between root and max level (2) 1`] = `
Object {
  "left": Object {
    "left": null,
    "right": null,
    "value": 1,
  },
  "right": Object {
    "left": null,
    "right": Object {
      "left": Object {
        "left": null,
        "right": null,
        "value": 50,
      },
      "right": null,
      "value": 150,
    },
    "value": 100,
  },
  "value": 2,
}
`;

exports[`insert into binary tree should insert node between root and max level 1`] = `
Object {
  "left": Object {
    "left": null,
    "right": null,
    "value": 1,
  },
  "right": Object {
    "left": Object {
      "left": null,
      "right": null,
      "value": 3,
    },
    "right": null,
    "value": 100,
  },
  "value": 2,
}
`;
